pylint/checkers/match_statements_checker.py

```diff
@@ -0,0 +1,54 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Match statement checker for Python code."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.interfaces import HIGH
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class MatchStatementChecker(BaseChecker):
+    name = "match_statements"
+    msgs = {
+        "E5000": (
```

Comment:
Above 49 is reserved for external plugin for now, right ?

Reply:
Indeed it's not very well documented https://github.com/pylint-dev/pylint/blob/13743cdd035765292d5f5f58274a21fb9aaafcf3/pylint/checkers/__init__.py#L5-L41

The script is here :https://github.com/pylint-dev/pylint/blob/main/script/get_unused_message_id_category.py